todo_interface.rb

```diff
@@ -0,0 +1,165 @@
+require_relative 'todo_item_class.rb'
+require_relative 'todo_list_class.rb'
+require 'colorize'
+require 'date'
+
+
+def input_date
+	while(true)
+		puts "Now enter the due date in the form \"dd-mm-yyyy\": "
+		date_input = gets.chomp
+		d, m, y = date_input.split('-')
+		if Date.valid_date? y.to_i, m.to_i, d.to_i
+			begin
+			 	 return Date.parse(date_input)
+			 rescue Exception => e
+			 end
+		end
+		puts "Oops! Invalid date -- try again!".red
+	end
+end
+
+def input_priority
+	while(true)
+		puts "Now enter priority. Remember: 1 = high, 2 = medium, 3 = low"
+		priority = gets.chomp
+		if (1..3).include? priority.to_i
+			return priority.to_i
+		end
+		puts "Oops! Invalid priority -- try again!".red
+	end
+end
+
+
+def initialize_list
+	puts "Welcome to your To-Do List! Let's get started!"
+
+	#initialize list
+	puts "To-Do List name: "
+	list_name = gets.chomp
+	list = ToDoList.new(list_name)
+	puts "Created To-Do List with name #{list.name}".blue
+	return list
```

Comment:
You don't need the `return` here, or almost anywhere. Ruby implicitly returns the last line of a method. The code will work fine with the return, but it is more common to see code without it.